sw/device/silicon_creator/rom/e2e/rom_e2e_self_hash_test.c

```diff
@@ -0,0 +1,57 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/crypto/include/hash.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey_memory.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  // Hash size.
+  kSha256HashSizeInBits = 256,
+  kSha256HashSizeInBytes = kSha256HashSizeInBits / 8,
+  kSha256HashSizeIn32BitWords = kSha256HashSizeInBytes / 4,
+};
+
+// TODO(#18686): replace with real golden hashes (when real keys are generated).
```

Comment:
Once we merge this, we will have to update this for every ROM change, right?
Can we add some instructions here that explain what needs to be done?

Reply:
Yep that is right. Sure thing, added some instructions.